lib/ansible/modules/database/misc/sqitch.py

```diff
@@ -0,0 +1,354 @@
+    description:
+      - Attach a list of a variables name and value for use by the database engine client when deploying, if it supports variables.
+        The format must be name=value, e.g., defuser='Homer Simpson'.
+        Overrides any values from set or loaded from the deploy.variables configuration. (rebase)
+    required: no
+'''
+
+EXAMPLES = '''
+- name: Deploy changes from sqitch.plan file located in /opt/sqitch to test_db database
+  sqitch:
+    command: deploy
+    plan_file: sqitch.plan
+    working_directory: /opt/sqitch
+    target: test_db
+
+- name: Revert all changes in test_db database
+  sqitch:
+    command: revert
+    target: test_db
+
+- name: Revert changes up to chng1 change in test_db database
+  sqitch:
+    command: revert
+    target: test_db
+    to_change: chng1
+
+- name: Rebase (revert all changes and deploy) test_db database from sqitch.plan file located in /opt/sqitch
+  sqitch:
+    command: rebase
+    plan_file: sqitch.plan
+    working_directory: /opt/sqitch
+    target: test_db
+    set:
+      - defuser='Homer Simpson'
+'''
+
+RETURN = '''
+#only defaults
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def get_target(cmd, target):
+    if target is not None:
+        cmd.append('--target')
+        cmd.append(target)
+
+
+def get_to_change(cmd, to_change):
+    if to_change is not None:
+        cmd.append('--to-change')
+        cmd.append(to_change)
+
+
+def get_verify(cmd, verify):
+    if verify:
+        cmd.append('--verify')
+    elif verify is not None and not verify:
+        cmd.append('--no-verify')
+
+
+def get_from_change(cmd, from_change):
+    if from_change is not None:
+        cmd.append('--to-change')
+        cmd.append(from_change)
+
+
+def get_mode(cmd, mode):
+    if mode is not None:
+        cmd.append('--mode')
+        cmd.append(mode)
+
+
+def get_set(cmd, set):
+    if len(set) > 0:
+        for variable in set:
+            cmd.append('--set')
+            cmd.append(variable)
+
+
+def get_log_only(cmd, log_only):
+    if log_only:
+        cmd.append('--log-only')
+
+
+def get_onto(cmd, onto):
+    if onto is not None:
+        cmd.append('--onto')
+        cmd.append(onto)
+
+
+def get_upto(cmd, upto):
+    if upto is not None:
+        cmd.append('--upto')
+        cmd.append(upto)
+
+
+def get_set_revert(cmd, set_revert):
+    if len(set_revert) > 0:
+        for variable in set_revert:
+            cmd.append('--set-revert')
+            cmd.append(variable)
+
+
+def get_set_deploy(cmd, set_deploy):
+    if len(set_deploy) > 0:
+        for variable in set_deploy:
+            cmd.append('--set-deploy')
+            cmd.append(variable)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            command=dict(default='deploy', choices=["deploy", "rebase", "revert", "status", "verify"]),
+            plan_file=dict(),
+            target=dict(),
+            engine=dict(),
+            working_directory=dict(),
```

Comment:
Ah, this needs to be set to to `type='path'` so it will be automatically expanded.

Reply:
That's the point. Now it works well and all tests are green ;)